### PR TITLE
Added POST /api/articles/:article_id/comments

### PR DESCRIPTION
**Pull request by JDean8**
Added POST /api/articles/:article_id/comments along with relevant integration tests

### CODE REVIEW

File: __tests__/api.test.js [hunk 1]
```diff
@@ -195,6 +195,55 @@ describe("GET /api/articles/:article_id/comments", () => {
...
+        username: "no_a_username",
+        body: "Valid comment",
+      })
+      .expect(400)
+      .then(({ body }) => {
+        expect(body.msg).toBe("Invalid username");
+      });
+  });
+});
+
```
Comment (NRMSMITH) [inline __tests__/api.test.js:246]: A few more tests to add here:
201 - ignores unnecessary properties on the sent comment
400 - missing required fields
400 - invalid id

**Review by NRMSMITH — CHANGES_REQUESTED** [review]
A few more tests to add please! :) Resubmit PR once done!

File: __tests__/api.test.js (hunk 1 above)
Comment (JDean8) [reply __tests__/api.test.js:246]: Updated to include the requested tests and will resubmit through NCHELP

File: __tests__/api.test.js [hunk 2]
```diff
@@ -195,6 +195,101 @@ describe("GET /api/articles/:article_id/comments", () => {
...
+        expect(comment).toEqual(
+          expect.objectContaining({
+            comment_id: expect.any(Number),
+            created_at: expect.any(String),
+            article_id: expect.any(Number),
+          })
+        );
+      });
+  });
+  test("Returns 201 and the comment when a comment is successfully added (ignoring redundant properties", () => {
```
Comment (B7Bprog) [inline __tests__/api.test.js:221]: How is this test checking if the response doesn't contain that extra property?

File: __tests__/api.test.js [hunk 3]
```diff
@@ -195,6 +195,101 @@ describe("GET /api/articles/:article_id/comments", () => {
...
+      .send({
+        username: "lurker",
+        body: "Usually don't comment",
+      })
+      .expect(404)
+      .then(({ body }) => {
+        expect(body.msg).toBe("No article found with that ID");
+      });
+  });
+  test("Should return a 404 and helpful message when clients tries to post comment for invalid article_id", () => {
```
Comment (B7Bprog) [inline __tests__/api.test.js:256]: Please make sure you have the right status code in your test description.

File: __tests__/api.test.js [hunk 4]
```diff
@@ -195,6 +195,101 @@ describe("GET /api/articles/:article_id/comments", () => {
...
+      .send({
+        username: "lurker",
+        body: "Usually don't comment",
+      })
+      .expect(400)
+      .then(({ body }) => {
+        expect(body.msg).toBe("Bad request, invalid type");
+      });
+  });
+  test("Should return a 400 and helpful message when clients tries to post comment for article with invalid username", () => {
```
Comment (B7Bprog) [inline __tests__/api.test.js:268]: This is not an invalid username, but one that is not in the database (yet). This should be a 404.

**Review by B7Bprog — CHANGES_REQUESTED** [review]
Hey Jack, a few things to refactor. Please see my comments. Once you are done, please send this PR back to us with the updates.

**Review by B7Bprog — APPROVED** [review]
Hey Jack, nice refactoring, please merge! 🎉

**Pull request merged**